tests/stress_tests/scripts/run_memcheck.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+# Copyright (C) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""
+This script runs memcheck tests isolated with help of gtest_parallel. It can
+upload memory measurment results to database and generate reports.
+"""
+
+import argparse
+from glob import glob
+import json
+import logging
+import os
+import subprocess
+import sys
+
+from memcheck_upload import create_memcheck_records, upload_memcheck_records, create_memcheck_report
+
+DATABASE = 'memcheck'
+COLLECTIONS = ["commit", "nightly", "weekly"]
+
+
+def run(args, log=None, verbose=True):
+    """ Run command
+    """
+    if log is None:
+        log = logging.getLogger('run_memcheck')
+    log_out = log.info if verbose else log.debug
+
+    log.info(f'========== cmd: {" ".join(args)}')  # pylint: disable=logging-format-interpolation
+
+    proc = subprocess.Popen(args,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.STDOUT,
+                            encoding='utf-8',
+                            universal_newlines=True)
+    output = []
+    for line in iter(proc.stdout.readline, ''):
+        log_out(line.strip('\n'))
+        output.append(line)
+        if line or proc.poll() is None:
+            continue
+        break
+    outs = proc.communicate()[0]
+
+    if outs:
+        log_out(outs.strip('\n'))
+        output.append(outs)
+    log.info('========== Completed. Exit code: %d', proc.returncode)
+    return proc.returncode, ''.join(output)
+
+
+def json_load(path_or_string):
+    """ Load json as file or as string
+    """
+    if os.path.isfile(path_or_string):
+        with open(path_or_string, 'r') as json_fp:
+            return json.load(json_fp)
+    else:
+        return json.loads(path_or_string)
+
+
+def main():
+    """Main entry point.
+    """
+
+    # remove additional args (arguments after --)
+    binary_args = []
+    for idx, arg in enumerate(sys.argv):
+        if arg == '--':
+            binary_args = sys.args[idx+1:].split()
+            sys.args = sys.args[:idx]
+            break
+
+    init_parser = argparse.ArgumentParser(add_help=False)
+    init_parser.add_argument('--timeline_report',
+                             help=f'Create timeline HTML report file name.')
+    init_parser.add_argument('--upload', action="store_true",
+                             help=f'Upload results to database.')
+    init_parser.add_argument('--gtest_parallel', help='Path to gtest-parallel to use.',
```

Comment:
@asomsiko should `gtest_parallel` placed in `init_parser`? Let's move it to `parser` (together with `binary` parsing)

Reply:
agree